doc: net: Add Thread protocol description (#13678)
zephyrproject-rtos/zephyr: doc/reference/networking/thread.rst

.. _thread_protocol_interface:

Thread protocol
###############

Overview
********
Thread is a low-power mesh networking technology, designed specifically for home
automation applications. It is an IPv6-based standard, which uses 6LoWPAN
technology over IEEE 802.15.4 protocol. IP connectivity allows to easily connect
a Thread mesh network to the internet with a Thread Border Router.

Thread specification provides a high level of network security. Mesh networks
built with Thread are secure - only authenticated devices can join the network
and all communications within the mesh are encrypted. More information about
Thread protocol can be found at
`Thread Group website <https://www.threadgroup.org>`_.

> Zephyr integrates an open source Thread protocol implementation called OpenThread.

Reviewer: How about:
```
Zephyr integrates an open source Thread protocol implementation called OpenThread,
documented on the `OpenThread website <https://openthread.io/>`_.
```

See `OpenThread website <https://openthread.io/>`_ for more information about
the OpenThread itself.

Internet connectivity

> =====================

Reviewer 2: I just wonder should the chapter use `*******` here, at least the other docs the Overview + other chapters are in the same level.

Author: Right, let's be consistent here.

A Thread Border Router is required to connect mesh network to the internet.

> An open source implementation of Thread Border Router is provided by OpenThread

Reviewer: ```suggestion
An open source implementation of Thread Border Router is provided by the OpenThread
```

community. See
`OpenThread Border Router guide <https://openthread.io/guides/border-router>`_

> for instructions on how to setup a Border Router.

Reviewer: "setup" is a noun, while "set up" is a verb phrase:
```suggestion
for instructions on how to set up a Border Router.
```

Sample usage

> ============

Reviewer 2: ditto

> You can try the OpenThread with Echo server and Echo client samples, which

Reviewer: ```suggestion
You can try using OpenThread with the Zephyr Echo server and Echo client samples, which
```
